.github/workflows/gradle.yaml

```diff
@@ -0,0 +1,33 @@
+name: Java CI
+
+on:
+  pull_request:
```

Comment:
You'll need to add branches to 'pull_request' event, otherwise this workflow will run on all pull requests.

Reply:
Unit tests should be fast, is it fine to run on all requests? 
If not, which branch do you suggest? and what format should it be added here.

Reply:
Ok, in that case it's fine to run it on all branches, we can leave it as is.